Sources/WhisperKit/MLX/MLXTextDecoder.swift

```diff
@@ -117,24 +126,59 @@ public final class MLXTextDecoder: TextDecoding {
         encoderOutputEmbeds: MLMultiArray,
         decoderKeyPaddingMask: MLMultiArray
     ) async throws -> (logits: MLMultiArray?, cache: DecodingCache?)? {
-        guard let decoder else {
+        let result = try await predictLogits(
+            inputIds: inputIds,
+            cacheLength: cacheLength,
+            keyCache: keyCache?.asMLXArray(FloatType.self),
+            valueCache: valueCache?.asMLXArray(FloatType.self),
+            kvCacheUpdateMask: kvCacheUpdateMask,
+            encoderOutputEmbeds: encoderOutputEmbeds.asMLXArray(FloatType.self).asMLXInput(),
+            decoderKeyPaddingMask: decoderKeyPaddingMask
+        )
+
+        guard let result = result,
+              let keyCacheResult = result.cache?.kvCache.map(\.k),
+              let valueCacheResult = result.cache?.kvCache.map(\.v)
+        else { return nil }
+
+        let keyCache = try? MLX.stacked(keyCacheResult).asMLMultiArray()
+        let valueCache = try? MLX.stacked(valueCacheResult).asMLMultiArray()
+        let decodingCache = DecodingCache(
+            keyCache: keyCache,
+            valueCache: valueCache,
+            alignmentWeights: nil
+        )
+
+        let logits = try? result.logits?.asMLMultiArray()
```

Comment:
seems like you've changed it to `try ?` in couple of places, is it intended?

Reply:
Good point, we want this to pass the throw up through the call stack. Reviewing build warnings as well.